src/nurse/models.py

```diff
@@ -66,3 +66,27 @@ def __str__(self):
     def is_valid(self):
         """Returns true if the prescription is still valid (e.g. hasn't expired)."""
         return self.start_date <= datetime.now().date() <= self.end_date
+
+    def get_prescriptions_expiring_soon():
+        today = datetime.now().date()
+        expiring_soon_date = today + timedelta(days=7)
+
+        prescriptions = Prescription.objects.filter(
+            end_date__lte=expiring_soon_date
+        ).filter(end_date__gte=today)
+
+        return prescriptions
+
+    def get_users_for_prescriptions(prescriptions):
+        users_associated = {}
+        for prescription in prescriptions:
+            patient = prescription.patient
+
+            nurse = patient.nurse_set.first()
+
+            if nurse:
+                user = nurse.user
+                if user:
+                    users_associated[prescription] = user
```

Comment:
I know this is a draft, but I have an early comment.
When there's a for loop within a DB query is often fishy.
Don't worry too much about it now, but after we've done writing test for this PR, we need to refactor this loop to leverage the DB